functional-samples/cookbook.sidepanel-action/service-worker.js

```diff
@@ -0,0 +1,22 @@
+async function setPanel() {
+  await chrome.sidePanel.setOptions({ path: 'sidepanel.html', enabled: true });
+  await chrome.sidePanel.setPanelBehavior({ openPanelOnActionClick: true });
+}
+
+async function clearPanel() {
+  await chrome.sidePanel.setOptions({ enabled: false });
+}
+// Enabled on google.com - disabled on all other sites
+const googleURL = 'https://www.google.com/';
+
+chrome.tabs.onActivated.addListener(() => {
```

Comment:
Given we have tab specific side panels now, how do you feel about the following approach:

- Listen to chrome.tabs.onUpdated instead of onActivated
- When the URL is a Google one, set the side panel for that tab ID
- When the URL isn't a Google one, disable the side panel for that tab ID

That is slightly nicer I think since it demonstrates the tab specific side panels and also avoids having to run code every time you switch tabs.

Reply:
Refactored. PTAL 👍